### ISSUE TITLE
Add Definitions of Relays, Relays+Caches, Relays+Replication Points, and Relays+Media Translators

### ISSUE THREAD

**Issue by SpencerDawkins** [fiestajetsam/draft-gruessing-moq-requirements#81]
From [the MOQ charter](https://datatracker.ietf.org/wg/moq/about/): 

> The working group will define MoQ so that the media publication protocol can leverage **coordinating relays, caches, or replication points** wherever applicable to improve the delivery performance.

I'm working from these assumptions:

- If there's a MOQ entity between client and server, that entity includes relay functionality
- In addition, a MOQ entity between client and server can include a cache
- In addition, a MOQ entity between client and server can include functionality for media media replication
- In addition, a MOQ entity between client and server can include functionality for transcoding media

The first three assumptions are explicitly mentioned in the charter.

It would be helpful to provide some level of detail about what is expected for these entities.

**Comment by SpencerDawkins** [comment]
From #81 - Expectations for relay functionality. 

At a minimum, we've had conversations about what level of granularity we expect relays to make forwarding decisions on, and how much we expect relays to understand about media, without introducing media-specific relay behaviour.

**Comment by SpencerDawkins** [comment]
@SpencerDawkins opinion - this isn't about drawing all the possible ways to combine these blocks - that would happen as part of discussion about deployment considerations (probably in other documents).

**Comment by fluffy** [comment]
I view Media Translators as a type of endpoint as they see some of all the media.

**Comment by SpencerDawkins** [comment]
@fluffy, 

> I view Media Translators as a type of endpoint as they see some of all the media.

I agree, if the Media Translator sees all of the media. If all Media Translators see all of the media (or, equally good, if we identify something that's sort of a Media Translator, that - I'm making this up as I go along - needs to see some, but not all, of the metadata that relays do not have access to), we can remove Media Translators from the list here, and consider that function as part of the systems engineering work that @hardie mentioned during the interim meeting yesterday.

**Comment by SpencerDawkins** [comment]
This issue was discussed at the MOQ interim meeting on 2023-01-31 - notes from that meeting are [here ](https://notes.ietf.org/notes-ietf-interim-2023-moq-03-moq?edit)- and again on 2023-02-01 - notes from that meeting are [here ](https://notes.ietf.org/notes-ietf-interim-2023-moq-05-moq?edit)- and are included here by reference.

Anyone working on a PR for this issue should take a look at the discussion notes.

**Comment by suhasHere** [comment]
@SpencerDawkins I can take a stab at the PR

**Comment by SpencerDawkins** [comment]
@suhasHere - thanks! You might want to keep an eye on discussion in the MOQ Entity design team about this, but if we can put something in the requirements draft that's enough to get us started on requirements, that would be helpful.

**Comment by SpencerDawkins** [comment]
What I would LIKE to do, is use the @suhasHerescenarios and architecture drafts (submitted before IETF 116), as a reference, rather than add anything in our document called "terminology" or "architecture". 

Not "add definitions", but "use terminology correctly".

**Comment by SpencerDawkins** [comment]
This issue is important, but depends on an agreed minimal architecture for a MOQ system, and secondarily, depends on some consensus for terminology in some wiki somewhere, to ensure that we don't produce multiple drafts with different terminology and different assumptions about what MOP systems look like. **These are both external dependencies that the MOQ working group has agreed matter in more than one formal meeting. How does MOQ make that happen?**

**Comment by SpencerDawkins** [comment]
The documents that we should be matching are [Media over QUIC Transport](https://datatracker.ietf.org/doc/draft-lcurley-moq-transport/) and [Media Over QUIC Media and Security Architecture](https://datatracker.ietf.org/doc/draft-nandakumar-moq-arch/). Ideally, they should also match each other, of course. :hand_over_mouth: ....
